### PR TITLE
Validator Manager SDK

### PR DESCRIPTION
**Pull request by sukantoraymond**
## Why this should be merged
Refactoring validator manager functionalities into sdk

## How this works
Refactoring existing code into SDK package

## How this was tested
Should pass Etna related CI

## How is this documented
NA

### CODE REVIEW

File: sdk/validatorManager/validator_manager.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	)
+}
+
+// PoaValidatorManagerGetPChainSubnetConversionWarpMessage constructs p-chain-validated (signed) subnet conversion warp
+// message, to be sent to the validators manager when
+// initializing validators set
+// the message specifies [subnetID] that is being converted
+// together with the validator's manager [managerBlockchainID],
+// [managerAddress], and the initial list of [validators]
+func PoaValidatorManagerGetPChainSubnetConversionWarpMessage(
```
Comment (michaelkaplan13) [inline sdk/validatorManager/validator_manager.go:111]: We should be consistent with our capitalization of PoA. I think either PoA or POA makes sense.

File: sdk/blockchain/blockchain.go [hunk 2]
```diff
@@ -306,3 +335,86 @@ func (c *Subnet) Commit(ms multisig.Multisig, wallet wallet.Wallet, waitForTxAcc
 	}
 	return tx.ID(), issueTxErr
 }
+
+// InitializeProofOfAuthority setups PoA manager after a successful execution of
+// ConvertSubnetTx on P-Chain
+// needs the list of validators for that tx,
+// [convertSubnetValidators], together with an evm [ownerAddress]
+// to set as the owner of the PoA manager
+func (c *Subnet) InitializeProofOfAuthority(
```
Comment (michaelkaplan13) [inline sdk/blockchain/blockchain.go:344]: Can we add unit tests for all of the new functions in the SDK package?

File: cmd/blockchaincmd/deploy.go [hunk 3]
```diff
@@ -484,7 +486,7 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
 	}
 
 	if sidecar.Sovereign {
-		if !convertOnly && !generateNodeID {
+		if !generateNodeID {
```
Comment (arturrez) [inline cmd/blockchaincmd/deploy.go:489]: why `!convertOnly` was removed?

File: sdk/validatormanager/validator_manager.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	ValidatorContractAddress = "0x5F584C2D56B4c356e7d82EC6129349393dc5df17"
+)
+
+var (
+	ErrAlreadyInitialized                  = errors.New("the contract is already initialized")
```
Comment (arturrez) [inline sdk/validatormanager/validator_manager.go:32]: nit: use `fmt.Errorf`

File: sdk/validatormanager/validator_manager.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+// manage validators on [subnetID], with
+// owner given by [ownerAddress]
+func PoAValidatorManagerInitialize(
+	rpcURL string,
+	managerAddress common.Address,
+	privateKey string,
+	subnetID ids.ID,
+	ownerAddress common.Address,
+) (*types.Transaction, *types.Receipt, error) {
+	const (
```
Comment (arturrez) [inline sdk/validatormanager/validator_manager.go:78]: nit: we should move it up so they are avaliable for sdk users to use

File: tests/e2e/testcases/validatormanager/suite.go [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+const (
+	CLIBinary         = "./bin/avalanche"
+	subnetName        = "e2eSubnetTest"
+	keyName           = "ewoq"
+	avalancheGoPath   = "--avalanchego-path"
```
Comment (arturrez) [inline tests/e2e/testcases/validatormanager/suite.go:34]: nit: what's the value of keeping it in constant and so `--avalanchego-path` is hidden behind it ?

**Review by arturrez — APPROVED** [review]

File: sdk/validatormanager/validator_manager.go (hunk 5 above)
Comment (arturrez) [reply sdk/validatormanager/validator_manager.go:78]: I meant constants

**Pull request merged**